src/main/java/TaskHandler.java

```diff
@@ -0,0 +1,144 @@
+import java.util.ArrayList;
+
+public class TaskHandler {
```

Comment:
Suggestion: Another way to improve this is to have a handler for each kinda task!